### PR TITLE
Dev

### LINKED ISSUE #90: Slow down on hover
Add effect of slowing down in a configurable radius around the mouse pointer

### PR DESCRIPTION
**Pull request by rubysdeadname**
PR for #90 
Added a parameter for the 'slow factor' which is a number between 0 and 1 that determines the slow-down of the particle when in the radius of the mouse. Also implemented linear scaling of speed from the mouse.

So if a particle had a speed of 10, the factor was set to 0.8, the radius was set to 40 and the particle was 20 away, the particle at that instant would have a speed of 10 * 0.8 * (20 / 40) = 4.

### CODE REVIEW

**Review by matteobruni — CHANGES_REQUESTED** [review]

File: src/Classes/Particle/Updater.ts [hunk 1]
```diff
@@ -45,13 +45,32 @@ export class Updater {
         const particle = this.particle;
 
         if (options.particles.move.enable) {
-            const moveSpeed = options.particles.move.speed / 2;
+            let moveSpeed = options.particles.move.speed / 2;
+            moveSpeed *= this.getProximitySpeedFactor();
```
Comment (gabrielbarker) [inline src/Classes/Particle/Updater.ts]: @matteobruni  In this later commit I've added this method to calculate the speed factor based on mouse proximity as well as the 'slow' options

File: src/Classes/Particle/Updater.ts [hunk 2]
```diff
@@ -45,14 +45,27 @@ export class Updater {
         const particle = this.particle;
 
         if (options.particles.move.enable) {
-            const moveSpeed = options.particles.move.speed / 2;
+            let moveSpeed = options.particles.move.speed / 2;
+            
+            if(options.interactivity.modes.slow.active && this.particleWithinMouseRadius()) {
+                moveSpeed *= options.interactivity.modes.slow.factor;
```
[the remark this answers is not in the document]
Comment (gabrielbarker) [reply src/Classes/Particle/Updater.ts]: I've added this feature in a later commit, I've linked you in a comment

**Review by matteobruni — APPROVED** [review]

File: src/Classes/Particle/Updater.ts (hunk 1 above)
Comment (matteobruni) [reply src/Classes/Particle/Updater.ts]: Thanks, it's perfect now!

Comment (gabrielbarker) [reply src/Classes/Particle/Updater.ts]: So glad I could help!

**Pull request merged**